addons/web_editor/static/src/scss/web_editor.common.scss

```diff
@@ -990,3 +998,11 @@ section, .oe_img_bg, [data-oe-shape-data] {
         margin-bottom: 1rem;
     }
 }
+
+code.o_inline_code {
```

Comment:
Not a big fan of the final result. What worries me the most is the the fact that this design makes text lines containing `code` to be have an huge vertical gap and to be consequentially rendered as "independent" from the others, almost like if they belongs to a paragraph ad-hoc (see picture, first paragraph). This message is indeed a bit too verbose just to prove that adding a `code` string here in github, doesn't  affect the paragraph line-height.. I think that we need something similar.

![image](https://github.com/odoo/odoo/assets/8558351/9ced88da-da36-4c43-b4d3-72632dc1c7d3)

To achieve the example in the second paragraph, I've briefly did this in my browser

```scss
    font-size: 85%;
    line-height: inherit;
    border: 0;
    padding: 0.2em 0.4em;
    background: var(--200);
    color: var(--900);
  ```
  
  _note_: this suggestion is `code.o_inline_code`.
  `pre` can have default font-size, border etc..